### PR TITLE
Added a negative test case - exclude key in array BA-5879

### PR DESCRIPTION
**Pull request by ichengchang**
Created a new unit test demonstrating negative results in key exlusion.

### CODE REVIEW

**Comment by ichengchang** [comment]
The test is to identify a scenario when exclude key does not work.

**Comment by danbills** [comment]
Yeah the ask was for a failing test, next time I think a "draft" PR is more appropriate

File: carbonite/src/main/scala/Carboniter.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+object Carboniter {
```
Comment (aednichols) [inline carbonite/src/main/scala/Carboniter.scala]: This looks like a stowaway from a related in-progress project

**Comment by ichengchang** [comment]
I added back the missing reference to "nested" node in the unit tests against Json array.

**Comment by ichengchang** [comment]
Wasn't paying attention to new defintion of COntrivedJsonWithArray, revert previous changes.  Everything looks good.

File: carbonite/src/main/scala/Carboniter.scala (hunk 1 above)
Comment (danbills) [reply carbonite/src/main/scala/Carboniter.scala]: ha!  whoops

**Comment by ichengchang** [comment]
JProfiler results on key exclusion operation

Screen Shot 2019-07-31 at 1 55 29 PM

**Comment by ichengchang** [comment]
Much better performance, 8-fold improvement

Screen Shot 2019-07-31 at 5 15 50 PM

File: core/src/main/scala/cromwell/util/JsonEditor.scala [hunk 2]
```diff
@@ -2,98 +2,67 @@ package cromwell.util
...
+  def includeJson(json: Json, keys: NonEmptyList[String]): Json = {
+    def folder: Folder[(Json, Boolean)] = new Folder[(Json, Boolean)] {
+      override def onNull: (Json, Boolean) = (Json.Null, false)
+      override def onBoolean(value: Boolean): (Json, Boolean) = (Json.fromBoolean(value), false)
+      override def onNumber(value: JsonNumber): (Json, Boolean) = (Json.fromJsonNumber(value), false)
+      override def onString(value: String): (Json, Boolean) = (Json.fromString(value), false)
+      override def onArray(value: Vector[Json]): (Json, Boolean) = {
+        val newArrayAndKeeps: immutable.Seq[(Json, Boolean)] = value.map(_.foldWith(folder))
+        val keep: Boolean = newArrayAndKeeps.map(_._2).foldLeft(false)(_ || _)
+        (Json.fromValues(newArrayAndKeeps.map(_._1)), keep)
```
Comment (kshakir) [inline core/src/main/scala/cromwell/util/JsonEditor.scala]: ToL-only: Generally I'm not a fan of the tuple `._2` syntax, preferring either custom classes or the verbose `case (_, descriptive) => descriptive`, but this seems to work.

Comment (danbills) [reply core/src/main/scala/cromwell/util/JsonEditor.scala]: yeah me neither :).  I"ll fix it

**Comment by ichengchang** [comment]
Latest JProfiler for Exclude keys operation

Screen Shot 2019-08-01 at 2 21 56 PM

**Comment by ichengchang** [comment]
JProfiler results for Include Keys

Screen Shot 2019-08-01 at 2 28 18 PM

**Comment by ichengchang** [comment]
Benchmarking results

::Benchmark JsonEditor with circe.includeExcludeJson(_, None, Some(NonEmptyList.of(<some exclude key>)))::
cores: 12
hostname: wm963-226
name: Java HotSpot(TM) 64-Bit Server VM
osArch: x86_64
osName: Mac OS X
vendor: Oracle Corporation
version: 25.211-b12
Parameters(Exclude keys traversal: json size MB -> 14): 22.716215929999994 ms
Parameters(Exclude keys traversal: json size MB -> 32): 222.96187330999993 ms

::Benchmark JsonEditor with circe.includeJson(_, NonEmptyList.one("message"))::
cores: 12
hostname: wm963-226
name: Java HotSpot(TM) 64-Bit Server VM
osArch: x86_64
osName: Mac OS X
vendor: Oracle Corporation
version: 25.211-b12
Parameters(Include keys traversal: json size MB -> 14): 23.666746470000003 ms
Parameters(Include keys traversal: json size MB -> 32): 151.92092569000005 ms

**Pull request merged**